docs/architecture/ADR-004-IBC-whitelist.md

```diff
@@ -0,0 +1,29 @@
+# ADR 004: IBC Whitelist/Allowlist
+<!-- disable markdownlint MD010 because Go code snippet uses tabs -->
+<!-- markdownlint-disable MD010 -->
```

Comment:
[nit] there is no Go code snipped in this file so this markdownlint disable isn't necessary

```suggestion
```